src/Assets/AssetHooks.cs

```diff
@@ -13,10 +19,39 @@ internal static class AssetHooks
 
         public static void Apply()
         {
+            // Static menu hooks building (select screen, dreams)
             IL.Menu.MenuIllustration.LoadFile_string += MenuIllustration_LoadFile_string;
             On.Menu.MenuScene.BuildScene += MenuScene_BuildScene;
+            
+            // Slideshow hooks (Intro and Outro are covered)
+            IL.Menu.SlugcatSelectMenu.StartGame += SlugcatSelectMenu_StartGame;
+            IL.RainWorldGame.ExitToVoidSeaSlideShow += RainWorldGame_ExitToVoidSeaSlideShow;
+            IL.Menu.SlideShow.ctor += SlideShow_ctor;
+            On.Menu.MenuScene.BuildScene += MenuScene_BuildScene_IntroOutroSlideshow;
+            On.Menu.SlideShowMenuScene.ctor += SlideShowMenuScene_ctor;
+
+            // Dream hooks
+            IL.RainWorldGame.Win += RainWorldGame_Win;
+            On.Menu.DreamScreen.SceneFromDream += SceneID_SceneFromDream;
+
+            // Clean the next Dream
+            On.Menu.MainMenu.ctor += MainMenu_ctor;
+            On.Player.Die += Player_Die;
         }
 
+        #region Clear the upcoming dream if the player exits the campaign or dies
+        private static void Player_Die(On.Player.orig_Die orig, Player self)
+        {
+            orig(self);
+            CustomScene.QueueDream("");
```

Comment:
What if the player dies but is revived, a player in coop dies, or a slugpup dies?